### PR TITLE
Activate the view-recipe from recipes menu

### LINKED ISSUE #93: Recipes on the main page should be clickable
Each recipe should redirect to its own view-recipe page.

### PR DESCRIPTION
**Pull request by GilShechter**
### Description

Each recipe in the recipes menu should be clickable and should redirect to its own page.

### What's new

- Changes the URL route to use the recipe title
- Modified the tests to fit the new code

Close #93

### CODE REVIEW

**Review by GalOved — APPROVED** [review]
Good work. Looking good for me.

**Review by efraimvis — DISMISSED** [review]

File: tasties/urls.py [hunk 1]
```diff
@@ -25,5 +25,5 @@
     path('login/', views.login_user, name="login"),
     path('register/', views.register, name="register"),
     path('logout/', views.logout_user, name="logout"),
-    path('view_recipe/', views.view_recipe, name="view_recipe")
+    path('view_recipe/<str:recipe_title>/', views.view_recipe, name="view_recipe")
```
Comment (natifridman) [inline tasties/urls.py:28]: Why not use the recipe id here instead?

Comment (liornoy) [reply tasties/urls.py:28]: +1

Comment (GilShechter) [reply tasties/urls.py]: Changes the URL to use the id instead of the title.
Added some more tests.

**Review by efraimvis — APPROVED** [review]

File: tasties/urls.py (hunk 1 above)
Comment (natifridman) [reply tasties/urls.py]: Great! think that you can't rely upon your titles will be always in English so using IDs in the URLs is always preferred

**Review by natifridman — APPROVED** [review]

**Comment by GilShechter** [comment]
Rebased and squashed my commits.

File: tasties_app/tests/test_recipes_view.py [hunk 2]
```diff
@@ -12,13 +12,26 @@ def test_recipes_view(self, client, signed_up_credentials):
...
         assert response.url == '/'
 
-    def test_view_recipe(self, client, signed_up_credentials):
+    def test_view_recipe(self, client, signed_up_credentials, recipes):
         client.post('/login/', data={'username': VALID_USER, 'password': VALID_PASSWORD})
-        response = client.get('/view_recipe/')
+        recipe = recipes[0]
+        response = client.get(f'/view_recipe/{recipe.id}/')
+        assert response.status_code == 200
+        assert response.request['PATH_INFO'] == f'/view_recipe/{recipe.id}/'
```
Comment (liornoy) [inline tasties_app/tests/test_recipes_view.py:20]: We should also check that the response includes the desired recipe object

**Review by liornoy — CHANGES_REQUESTED** [review]
added a small comment regarding the `test_view_recipe` .
Also, what is the difference between the invalid_id test and the not_found test?

**Comment by GilShechter** [comment]
> added a small comment regarding the `test_view_recipe` . Also, what is the difference between the invalid_id test and the not_found test?

Each `recipe.id` is an integer, so in `invalid_id` I inserted a string `test1` instead of an integer and expected to get a 404 error.
In `not_found` I insert a valid id that doesn't exist in the database and expect to be redirected back to the recipes page.

**Comment by liornoy** [comment]
> > added a small comment regarding the `test_view_recipe` . Also, what is the difference between the invalid_id test and the not_found test?
> 
> Each `recipe.id` is an integer, so in `invalid_id` I inserted a string `test1` instead of an integer and expected to get a 404 error. In `not_found` I insert a valid id that doesn't exist in the database and expect to be redirected back to the recipes page.

ok, thanks for the explanation.

**Comment by liornoy** [comment]
LGTM

**Review by liornoy — APPROVED** [review]

**Pull request merged**